workers/generic-worker/errorreport/errorreport_test.go

```diff
@@ -0,0 +1,101 @@
+package errorreport
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/taskcluster/taskcluster/v30/internal/workerproto"
+	workerProtoTesting "github.com/taskcluster/taskcluster/v30/internal/workerproto/testing"
+)
+
+func setupProtocols() (*workerproto.Protocol, *workerproto.Protocol) {
+	workerTransp, runnerTransp := workerProtoTesting.NewLocalTransportPair()
+	workerProto := workerproto.NewProtocol(workerTransp)
+	workerProto.AddCapability("error-report")
+	workerProto.Start(true)
+
+	runnerProto := workerproto.NewProtocol(runnerTransp)
+	runnerProto.AddCapability("error-report")
+	return workerProto, runnerProto
+}
+
+func TestSendEmptyExtra(t *testing.T) {
+	errorReported := false
+	lock := sync.Mutex{}
+
+	workerProto, runnerProto := setupProtocols()
+
+	runnerProto.Register("error-report", func(msg workerproto.Message) {
+		assert.Contains(t, msg.Properties, "title")
+		assert.Equal(t, "generic-worker error", msg.Properties["title"])
+		assert.Contains(t, msg.Properties, "kind")
+		assert.Equal(t, "worker-error", msg.Properties["kind"])
+		assert.Contains(t, msg.Properties, "description")
+		assert.Equal(t, "test error", msg.Properties["description"].(string))
+		assert.Contains(t, msg.Properties, "extra")
+		assert.Equal(t, msg.Properties["extra"], map[string]interface{}{})
+		errorReported = true
+		lock.Unlock()
+	})
+
+	runnerProto.Start(false)
+	runnerProto.WaitUntilInitialized()
+
+	// unlocked in callback
+	lock.Lock()
+	// report test error
+	Send(workerProto, fmt.Errorf("test error"), nil)
+
+	lock.Lock()
+	for i := 0; i < 200; i++ {
+		if !errorReported {
+			time.Sleep(time.Millisecond * 10)
+		} else {
+			return
+		}
+	}
+	lock.Unlock()
+	t.Fatalf("No error-report was received")
```

Comment:
Since you've acquired the lock on line 52, the callback must have completed, so no need to have the waiting `for` loop on line 53. The lock should be released independently of whether the test passes or not, so it is probably simplest to release it immediately after acquiring it.

Nit: the only way the test can fail is if it hangs waiting to acquire the lock on line 52, which would happen when the test eventually times out. Perhaps a context with a deadline would provide quicker feedback, although maybe not worth changing now.

```suggestion
	lock.Unlock()
	assert.True(t, errorReported, "No error-report was received")
```